### PR TITLE
[RFC] Removed the web_profiler.position config option

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | yes |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

In the past we needed this option because the profiler displayed the toolbar on the top. Right now this option is only used by end-users. When Symfony 2 was published, this option was useful because symfony 1 displayed the toolbar on the top and therefore, some developers were mad at the new toolbar which was displayed at the bottom. Nowadays everybody is happy with the bottom toolbar.

### CODE REVIEW

File: src/Symfony/Bundle/WebProfilerBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -35,6 +35,14 @@ public function getConfigTreeBuilder()
         $rootNode = $treeBuilder->root('web_profiler');
 
         $rootNode
+            ->beforeNormalization()
+                ->ifTrue(function ($v) { return isset($v['position']); })
```
Comment (stof) [inline src/Symfony/Bundle/WebProfilerBundle/DependencyInjection/Configuration.php]: should be ``array_key_exists`` IMO, to account for ``null``. They should not configure it at all

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar.html.twig [hunk 2]
```diff
@@ -19,7 +19,7 @@
         </a>
     </div>
     <style>
-        {{ include('@WebProfiler/Profiler/toolbar.css.twig', { 'position': position, 'floatable': true }) }}
+        {{ include('@WebProfiler/Profiler/toolbar.css.twig', { 'floatable': true }) }}
```
Comment (stof) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar.html.twig]: is there any need for this ``floatable`` flag for the CSS ?

Comment (javiereguiluz) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar.html.twig]: This option is no longer used. Removed. Thanks.

File: src/Symfony/Bundle/WebProfilerBundle/DependencyInjection/Configuration.php (hunk 1 above)
Comment (javiereguiluz) [inline src/Symfony/Bundle/WebProfilerBundle/DependencyInjection/Configuration.php]: You are right. Done. Thanks.

**Comment by WouterJ** [comment]
:-1: This option still makes sense in newer Symfony versions.

The toolbar uses fixed positioning to always appear on the bottom of the screen. This means it overlaps with any other bar that's configured at the bottom. For instance, some designs use a bottom nav bar. This means the toolbar always overlaps the nav bar. In such cases, it's very usefull to be able to position the toolbar at the top of the page instead.

File: src/Symfony/Bundle/WebProfilerBundle/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -35,6 +35,14 @@ public function getConfigTreeBuilder()
         $rootNode = $treeBuilder->root('web_profiler');
 
         $rootNode
+            ->beforeNormalization()
+                ->ifTrue(function ($v) { return array_key_exists('position', $v); })
```
Comment (stof) [inline src/Symfony/Bundle/WebProfilerBundle/DependencyInjection/Configuration.php]: you can put the ``beforeNormalization`` in the ``position`` node directly, with a ``->always()`` condition

**Comment by fabpot** [comment]
@WouterJ comment makes sense to me.

**Comment by fabpot** [comment]
By the way, I think it was added exactly for @WouterJ use case, not because people were angry about the new position (AFAIR, people didn't complain back then).

**Pull request closed without merging**